modules/renter/contractor/contractor.go

```diff
@@ -17,6 +17,11 @@ var (
 	errNilTpool  = errors.New("cannot create contractor with nil transaction pool")
 )
 
+type cachedRevision struct {
+	revision    types.FileContractRevision
+	merkleRoots []crypto.Hash
```

Comment:
for now this is fine, but we should note that merkleRoots may end up being very large, dozens or hundreds of MB depending on how many sectors have been uploaded. Long term, this should probably be a diff over the existing saved merkleRoots.